tests/django_pg_migration_tools/test_operations.py

```diff
@@ -1755,3 +1756,701 @@ def test_when_valid_constraint_and_alter_table_already_performed(self):
+    def test_when_not_allowed_to_migrate_by_the_router(self):
+        project_state = ProjectState()
+        project_state.add_model(ModelState.from_model(IntModel))
+        new_state = project_state.clone()
+        operation = operations.SaferAddFieldForeignKey(
+            model_name="intmodel",
+            name="char_model_field",
+            field=models.ForeignKey(CharModel, null=True, on_delete=models.CASCADE),
+        )
+
+        with connection.schema_editor(atomic=False, collect_sql=False) as editor:
+            with utils.CaptureQueriesContext(connection) as queries:
+                operation.database_forwards(
+                    self.app_label, editor, project_state, new_state
+                )
+        # No queries have run, because the migration wasn't allowed to run by
+        # the router.
+        assert len(queries) == 0
+
+        # Try the same for the reverse operation:
+        with connection.schema_editor(atomic=False, collect_sql=False) as editor:
+            with utils.CaptureQueriesContext(connection) as queries:
+                operation.database_backwards(
+                    self.app_label, editor, project_state, new_state
+                )
+
+        # No queries have run, because the migration wasn't allowed to run by
+        # the router.
+        assert len(queries) == 0
+
+    @pytest.mark.django_db(transaction=True)
+    def test_operation(self):
+        with connection.cursor() as cursor:
+            # Set the lock_timeout to check it has been returned to
+            # its original value once the fk index creation is completed.
+            cursor.execute(_SET_LOCK_TIMEOUT)
+
+        project_state = ProjectState()
+        project_state.add_model(ModelState.from_model(IntModel))
+        new_state = project_state.clone()
+        operation = operations.SaferAddFieldForeignKey(
+            model_name="intmodel",
+            name="char_model_field",
+            field=models.ForeignKey(CharModel, null=True, on_delete=models.CASCADE),
+        )
+
+        assert operation.describe() == (
+            "Add field char_model_field to intmodel. Note: Using "
+            "django_pg_migration_tools SaferAddFieldForeignKey operation."
+        )
+
+        with connection.schema_editor(atomic=False, collect_sql=False) as editor:
+            with utils.CaptureQueriesContext(connection) as queries:
+                operation.database_forwards(
+                    self.app_label, editor, project_state, new_state
+                )
+        assert len(queries) == 9
+
+        assert queries[0]["sql"] == dedent("""
+            SELECT 1
+            FROM pg_catalog.pg_attribute
+            WHERE
+                attrelid = 'example_app_intmodel'::regclass
+                AND attname = 'char_model_field';
+        """)
+        assert queries[1]["sql"] == dedent("""
+            ALTER TABLE "example_app_intmodel"
+            ADD COLUMN IF NOT EXISTS "char_model_field"
```

Comment:
This SQL illustrates a name issue that the code has.

We're asking to add a FK field named `char_model_field`. By convention, Django implements that field using a column with `_id` appended, i.e. `char_model_field_id`.

The sample SQL in the PR description has `bar_id` for the name of the column that's added. I think to achieve that with the current code you'd need to do:

```python
operations.SaferAddFieldForeignKey(
    model_name="foo",
    name="bar_id",  # This would normally be just "bar"
    field=models.ForeignKey(Bar, null=True, on_delete=models.CASCADE),
)
```

I've put this comment on this test because I'm not sure where in the code the change would need to be made.

Reply:
Good catch. We should keep this consistent with Django.
Fixup: 240a7cfe0d5fd6af52bdb8bafcfcd426ec19cd23